src/features/soft_navigations/aggregate/index.js

```diff
@@ -0,0 +1,163 @@
+import { getConfigurationValue } from '../../../common/config/config'
+import { handle } from '../../../common/event-emitter/handle'
+import { registerHandler } from '../../../common/event-emitter/register-handler'
+import { HarvestScheduler } from '../../../common/harvest/harvest-scheduler'
+import { single } from '../../../common/util/invoke'
+import { timeToFirstByte } from '../../../common/vitals/time-to-first-byte'
+import { FEATURE_NAMES } from '../../../loaders/features/features'
+import { AggregateBase } from '../../utils/aggregate-base'
+import { FEATURE_NAME, INTERACTION_STATUS } from '../constants'
+import { AjaxNode } from './ajax-node'
+import { InitialPageLoadInteraction } from './initial-page-load-interaction'
+import { Interaction } from './interaction'
+
+export class Aggregate extends AggregateBase {
+  static featureName = FEATURE_NAME
+  constructor (agentIdentifier, aggregator) {
+    super(agentIdentifier, aggregator, FEATURE_NAME)
+
+    const harvestTimeSeconds = getConfigurationValue(agentIdentifier, 'spa.harvestTimeSeconds') || 10
+    this.interactionsToHarvest = []
+    this.interactionsAwaitingRetry = []
+
+    this.scheduler = new HarvestScheduler('events', {
+      onFinished: this.onHarvestFinished.bind(this),
+      retryDelay: harvestTimeSeconds
+    }, { agentIdentifier, ee: this.ee })
+    this.scheduler.harvest.on('events', this.onHarvestStarted.bind(this))
+
+    this.initialPageLoadInteraction = new InitialPageLoadInteraction(agentIdentifier)
+    timeToFirstByte.subscribe(({ entries }) => {
+      const loadEventTime = Math.round(entries[0].loadEventEnd)
+      this.initialPageLoadInteraction.finish(loadEventTime)
+      this.interactionsToHarvest.push(this.initialPageLoadInteraction)
+      this.initialPageLoadInteraction = null
+      this.scheduler.startTimer(harvestTimeSeconds, 0.1) // give buffered (ajax & jserror) events some time to settle before sending the initial page load
+    })
+
+    this.interactionInProgress = null // aside from the "page load" interaction, there can only ever be 1 ongoing at a time
+
+    this.blocked = false
+    registerHandler('block-spa', () => { // if rum response determines that customer lacks entitlements for spa endpoint, this feature shouldn't harvest
```

Comment:
This should probably be implemented in the opposite way, where it waits for the 'spa' flag and then initializes.  Could be a race between harvesting the initial ixn and the block-spa flag